### DISCUSSION TITLE
iOS?

### DISCUSSION THREAD

**Discussion by LeslieOA** [digidem/mapeo-mobile#762] [Ideas]
I really should have searched the old issues first, but: does this project need any *active* assistance with an iOS build?

Again: I've yet to attempt to install/link/deploy, but would love to help.

Thanks.

**Comment by gmaclennan** [comment]
Hi @LeslieOA thanks for getting in touch about this. We've just completed a refactor of the Mapeo Mobile code and active assistance with an iOS build would be really great. I did some early builds with a skeletal project that included the peer-to-peer backend (we run a node process on the phone using [nodejs-mobile-react-native](https://github.com/janeasystems/nodejs-mobile-react-native)) and that worked on iOS ok. However as I added react-native dependencies iOS stopped working, and I don't have the iOS knowledge to know what went wrong. I think some of our native deps require a podfile, and I'm not sure how that works. I prioritised the Android build just for time issues.

If you have any time to have a go at getting a build working for iOS that would be _amazing_. Whilst not many of our partners in South America use iOS devices, we recently ran some trainings at the [indigenous mapping workshop](https://www.indigenousmaps.com/2018imw/) in Montreal, Canada, and there was a lot of interest from First Nations participants there, many of whom use iPhones. Getting this working on iOS would be great for raising awareness of the app in North America / Europe where a lot more people seem to have iPhones, particularly potential funders :)

**Comment by okdistribute** [comment]
cc @staltz who is also currently working on iOS support with manyver.se, and uses a similar stack to mapeo.

**Comment by staltz** [comment]
To help a bit, since Karissa pinged me, I'll add that iOS support should be theoretically easy because nodejs-mobile-react-native was originally built for iOS and only later added Android support, so it's fair to say that it aims to support iOS first-class.

The specific obstacles to getting the app compiling and running on iOS will depend on the other dependencies used: (1) react native libraries sometimes are built for only iOS or only Android, sometimes both, so that in the only-Android case you might need to improve the library yourself, (2) native modules used in the node.js backend might have compilation obstacles related to the build tools on macOS (XCode) or the target architecture (arm64).

**Comment by LeslieOA** [comment]
Thanks @gmaclennan @karissa; I hope (that eventually) my assistance is useful. Love the work that  you're doing!
@staltz I've been meaning to signup to Manyverse for a hot minute. Done.

**@.everyone**: I see what you mean. I can understand the priority being Android as it's more populous. It'd be nice if there were more low cost B2G/KaiOS devices in the wild for a "HTML5+" version/port.

I'll be taking a look into things over the next week. Let me know if there's a place you all communicate for (potential) contributors aside/outside of GH Issues.

Thanks.

**Comment by gmaclennan** [comment]
hi @LeslieOA Sorry I didn't come back to this for a while, we were working hard towards the most recent beta release for a field trip to Peru this week and a training for some communities to start using the app.

For communicating with contributors and the team, the best place is [#ddem on freenode IRC](http://webchat.freenode.net/?channels=ddem) or the mirrored [#ddem channel on Slack](https://slack.digital-democracy.org).

If you are based in London and have the freedom to work outside the office, I'd be happy to meet up and co-work one day. I'm in the north-east, and generally work from home and cafes.

**Comment by LeslieOA** [comment]
Not a problem @gmaclennan! I've been busy also, but popped back here to reference this project in another issue on https://github.com/digidem/osm-p2p-db/.

I am London based and work in Central but live East, let's catch up sometime. I'll add the above if I haven't already.
